### DISCUSSION TITLE
How to Automatically Re-Execute a Specific Rule on Tab Activation in VSCode

### DISCUSSION THREAD

**Discussion by SyMind** [eslint/eslint#17877] [Q&A] [answered]
# Background

I've created an ESLint custom rule to enforce the use of decorators from `class-validator` in DTO (Data Transfer Object) classes. The rule works by first identifying DTO classes referenced within controller classes. For example, in the `UserController`, the `GetUserRequest` class is identified as a DTO as shown below:

```js
// userController.ts
import { GetUserRequest } from './dto';

@Controller()
class UserController {
  @Get('/user')
  getUser(req: GetUserRequest) {
    // Function body
  }
}
```

After such a determination is made, I intend the rule to actively check the `GetUserRequest` class within `dto.ts` for the appropriate use of decorators.

# Problem

The issue arises under the following scenario:

1. The `GetUserRequest` class already exists within `./dto.ts`.
2. I add code to `UserController.getUser()` to use the `GetUserRequest` DTO.
3. When I return to the `dto.ts` file tab in VS Code, the custom rule does not re-trigger to recheck the file for decorator use.

Currently, the re-evaluation of the file doesn't seem to occur when the tab regains focus in VSCode. I'm looking for a way to ensure the rule re-runs in such cases to verify the enforcement of decorators in the DTO definition.

**Comment by Rec0iL99** [comment, marked as answer]
Hi @SyMind, the [`vscode-eslint`](https://github.com/Microsoft/vscode-eslint) extension is not maintained by the ESLint team. Please consult the maintainers there for help.

**Reply by SyMind** [reply]
Thank you. I have opened an issue on the vscode-eslint GitHub repository. https://github.com/microsoft/vscode-eslint/issues/1759
